### ISSUE TITLE
[4.2+ issue] OSError: cannot identify image file <ImageFieldFile: images/40_3ez0uO3>

### ISSUE THREAD

**Issue by fish-ball** [python-pillow/Pillow#2628]
After upgraded to 4.2.1, I got the error:

> OSError: cannot identify image file <ImageFieldFile: images/40_3ez0uO3>

It was actually because of the file exension, if I rename the file to `*.jpg`, or down grade to 4.1.1, the issue was solved.

```python
# working with django:

from PIL import Image
member = Member.objects.get(id=42)
Image.open(member.image)  # image is a `models.ImageField`, which raises the error above.
```

**Comment by wiredfool** [comment]
Can you post a small repro that includes the image, but does not rely on django?

**Comment by weijarz** [comment]
@wiredfool 
I got same error (python 2.7), e.g.

save `http://bos.nj.bpc.baidu.com/v1/mediaspot/41afc86b9243628f3b0aa16c546e6668.jpeg` to `/tmp/a/err.jpeg`

```
>>> from PIL import Image
>>> Image.open('/tmp/a/err.jpeg')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/wjzhou/ws/kindle4rss/venv/lib/python2.7/site-packages/PIL/Image.py", line 2519, in open
    % (filename if filename else fp))
IOError: cannot identify image file '/tmp/a/err.jpeg'
```

**Comment by fish-ball** [comment]
@wiredfool It can reproduce by opening a jpg file which is renamed to a filename with no extension.

**Comment by wiredfool** [comment]
@fish-ball I can't reproduce it with a known good image.  (here, hopper_noext is hopper.jpg from the test suite)

```
>>> from PIL import Image
>>> im = Image.open('hopper_noext')
>>> im.show()
>>> Image.PILLOW_VERSION
'4.2.1'
```

I'm suspicious here, because:
1) There's no dependence on filenames in Pillow's open function, as it works just as well from a BytesIO
2) There's no small reproduction. 


@weijarz Your image doesn't work on 4.1.1 or 4.2.0, even including the extension, so it's a different issue.

**Comment by wiredfool** [comment]
@weijarz Your issue is due to an invalid exif segment.

**Comment by hugovk** [comment]
Closing this as PR https://github.com/python-pillow/Pillow/pull/2632 has been merged. Let us know if this is still a problem with latest master and we can reopen.

**Comment by jdhao** [comment]
@hugovk ,I have also met this issue on CentOS, the PIL version is 4.2.1. On my Windows machine, the PIL version is 3.42 and it can open the image without problem. Then I updated the PIL on Windows to 4.2.1, the same error message occured.

**Comment by hugovk** [comment]
This works for me on macOS Sierra:
```python
from PIL import Image, PILLOW_VERSION
print(PILLOW_VERSION)
im = Image.open("2628.jpg")

im.show()
print(im)
```
```bash
[hugo:/private/tmp] % python2 --version
Python 2.7.14

[hugo:/private/tmp] % python2 2628.py
4.3.0
<PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=533x251 at 0x105541050>
```
```bash
⌂85% [hugo:/private/tmp] % python3 --version
Python 3.6.3

⌂77% [hugo:/private/tmp] % python3 2628.py
4.3.0
<PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=533x251 at 0x101737940>
```
```
⌂74% [hugo:/private/tmp] % md5 2628.jpg
MD5 (2628.jpg) = 96b03ae60e959511d542668a7ad3d871

⌂62% [hugo:/private/tmp] 1 % shasum 2628.jpg
ae364fd303aa1002a9e6ebfc86613999a030de19  2628.jpg
```

**Comment by jdhao** [comment]
I updated the pillow version using the [conda-forge channel](https://anaconda.org/conda-forge/pillow), now the error is gone.

**Comment by scythargon** [comment]
still happens with this image for example
http://asburyplace.org/wp-content/uploads/2017/01/AL-Slide-diversity.jpg

**Comment by thebeancounter** [comment]
anything new about that? still happening to me with PIL 5.4.1

**Comment by lggwettmann** [comment]
Same here with 6.2.0

**Comment by radarhere** [comment]
@lggwettmann can you provide a self-contained example, ideally that does not rely on any third-party software, with an image?

**Comment by lggwettmann** [comment]
Oh sorry, @radarhere - we use it with ReportLab (and solved it with a workaround) and I don't have time to provide a self-contained example now. 🙈 I highly appreciate your help though!

**Comment by wutonghhu** [comment]
> Same here with 6.2.0

Have you solved the problme? my pillow version is 7.2.1

**Comment by radarhere** [comment]
@wutonghhu can you provide a self-contained example and an image that this is happening for?
